src/uuid/yaml.cr

```diff
@@ -0,0 +1,48 @@
+require "yaml"
+require "uuid"
+
+# Creates UUID from YAML using `YAML::ParseContext`.
+#
+# NOTE: `require "uuid/yaml"` is required to opt-in to this feature.
+#
+# ```
+# require "yaml"
+# require "uuid"
+# require "uuid/yaml"
+#
+# class Example
+#   include YAML::Serializable
+#
+#   property id : UUID
+# end
+#
+# example = Example.from_yaml("uuid: 50a11da6-377b-4bdf-b9f0-076f9db61c93")
+# example.id # => UUID(50a11da6-377b-4bdf-b9f0-076f9db61c93)
+# ```
+def UUID.new(ctx : YAML::ParseContext, node : YAML::Nodes::Node)
+  ctx.read_alias(node, String) do |obj|
+    return UUID.new(obj)
+  end
+
+  if node.is_a?(YAML::Nodes::Scalar)
+    value = node.value
+    ctx.record_anchor(node, value)
+    UUID.new(value)
+  else
+    node.raise "Expected String, not #{node.kind}"
+  end
+end
+
+struct UUID
```

Comment:
Since `UUID` needs to be reopened anyways, you could move the constructor method into the namespace as well.

Reply:
Done!